lstchain/reco/utils.py

```diff
@@ -862,3 +863,49 @@ def get_events_in_GTI(events, CatB_cal_table):
     gti_mask = gti[events['calibration_id']]
 
     return events[gti_mask]
+
+def compute_rf_event_weights(events):
+    """
+    Compute event-wise weights. Can be used for correcting for the different
+    statistics present in each pointing node of the MC training sample,
+    to avoid "jumps" in the performance of the random forests
+
+    events: a DL1 parameters dataframe
+    """
+
+    # Add a 'weight' column to the input table
+    weights = np.array(np.ones(len(events)))
+
+    # First identify existing telescope pointings in the sample:
+    # Convert to degrees and round to avoid potential
+    # rounding issues:
+    alt = np.round(events['alt_tel'], decimals=5)
+    az = np.round(events['az_tel'], decimals=5)
+    pointings = np.unique([alt, az], axis=1).T
+
+    # Find the total statistics in each of the pointings:
+    stats = []
+    for tel_alt_az in pointings:
+        mask = (np.isclose(tel_alt_az[0], events['alt_tel'],
+                           atol=1e-5, rtol=0) &
+                np.isclose(tel_alt_az[1], events['az_tel'],
+                           atol=1e-5, rtol=0))
+        stats.append(mask.sum())
+
+    stats = np.array(stats)
+    weight_per_pointing = stats.mean() / stats
+
+    # Now set the weights.
+    # Weight in a given node will be mean_events_per_node / n_events_in_node
+
+    for ipointing, tel_alt_az in enumerate(pointings):
+        mask = (np.isclose(tel_alt_az[0], events['alt_tel'],
+                           atol=1e-5, rtol=0) &
+                np.isclose(tel_alt_az[1], events['az_tel'],
+                           atol=1e-5, rtol=0))
+        weights[mask] = weight_per_pointing[ipointing]
+
+    events['weight'] = weights
```

Comment:
If the dataframe already has a column called `weights`, it is silently replaced. It should probably raise a ValueError in this case.

Reply:
I added a warning, which I think is enough (and we do not write out this column anywhere, so it would have to be a different 'weight')